src/instructlab/model/train.py

```diff
@@ -91,28 +175,87 @@
     ),
 )
 @click.option(
-    "--model-name",
-    default="instructlab/merlinite-7b-lab",
-    show_default=True,
-    help="model name to use in training",
+    "--max-seq-len", type=int, help="maximum length, in tokens, of a single sample."
+)
+@click.option("--max-batch-len", type=int)
+@click.option(
+    "--effective-batch-size", type=int, help="total batch size across all GPUs"
+)
+@click.option(
+    "--save-samples",
+    type=int,
+    help="The number of samples processed in between checkpoints.",
+)
+@click.option("--learning-rate", type=float, help="learning rate for training")
+@click.option("--warmup-steps", type=int, help="warmup steps for training")
+@click.option(
+    "--cpu-offload-optimizer", type=bool, help="if true enables optimizer offload"
+)
+@click.option("--cpu-offload-ratio", type=float, help="cpu offload optimizer ratio")
+@click.option(
+    "--cpu-offload-optimizer-pin-memory",
+    type=bool,
+    help="if true pin memory when using cpu optimizer",
+)
+# below flags are invalid if lora == false
+@click.option("--rank", type=int, help="rank of update matricies")
+@click.option("--alpha", type=float, help="how influential/strong lors tune will be")
+@click.option("--dropout", type=float, help="dropout for LoRA layers")
+@click.option(
+    "--target-modules",
+    multiple=True,
+    default=[],
+)
+@click.option("--quantize-data-type", type=str, default=None)
+@click.option(
+    "--is-padding-free",
+    type=bool,
+    help="whether or not we are training a padding free transformer.",
+)
+@click.option(
+    "--gpus",
+    "nproc_per_node",
+    type=int,
+    help="this is the number of GPUs to use. This is a torch specific arg and must be called nproc-per-node",
+)
+@click.option("--nnodes", type=int, help="number of machines in the training pool.")
```

Comment:
for the CLI I think we should remove `--nnodes`, `--node-rank`, `--rdzv-id`, and sensibly default `---rdzv-endpoint` because we aren't supporting multi-node torchrun. Those configuration are purely for multi-proc + multi-node.

Reply:
I honestly think providing all flags and letting the library decided what they are going to allow in terms of options is the right move but I don't feel too strongly here